fetchai/ledger/genesis/genesis.py

```diff
@@ -0,0 +1,122 @@
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+import json
+import hashlib
+import base64
+import base58
+import time
+
+from fetchai.ledger.crypto import Entity
+
+def calc_resource_id(resource_address):
```

Comment:
Check with @NelisDrost this might be duplicated in other parts of the library

Reply:
There isn't a good existing place to import this from, though there are other places we do the same calculation - we should think about refactoring this.

Also, is the output of this function strictly a resource ID - I thought it was the sha256 hash that was the resource ID, not the b64 encoding of that hash?

Reply:
This will almost certainly get refactored in another PR, but looks good for now.